### PR TITLE
[DOM-48532] Domino Cost App

### PR DESCRIPTION
**Pull request by ddl-juan-cistaro**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,79 @@
-# app-costs-dashboard
\ No newline at end of file
+# Domino Cost App
+
+This describes the how to setup and run `Domino Cost App`.
```
Comment (ddl-olsonJD) [inline README.md:3]: This repo describes how to setup and run the `Domino Cost App` along with a Domino Cost App that can be used as a starting point.

File: README.md [hunk 2]
```diff
@@ -1 +1,79 @@
...
+_Table of Contents_
+
+- [Create New Project (optional)](#create-new-project-optional)
+- [Publish the App ](#publish-the-app)
+  - [Checking running status](#checking-running-status-optional)
+- [Accessing Domino Cost App](#accessing-domino-cost-app)
+
+# Create New Project (optional)
+
+To deploy Domino Cost app we need a project to be a home for it. We recomend creating a specific project for it.
```
Comment (ddl-olsonJD) [inline README.md:14]: The Domino Cost app requires a project to be launched from.

File: README.md [hunk 3]
```diff
@@ -1 +1,79 @@
...
+
+1. Go to the project's dashboard and click on `New Project`
+
+   ![project's dashboard](/img/01.projectsDashboard.png)
+
+2. Select a name and visibility that you prefer for this project.
```
Comment (ddl-olsonJD) [inline README.md:20]: Select a name and visibility for this project.

File: README.md [hunk 4]
```diff
@@ -1 +1,79 @@
...
+1. Download the files in the [/app](/app) folder and go to the `Code` section by clicking in the side bar of the project. There, upload the three files into the project:
+
+   ![upload files](/img/03.uploadFiles.png)
+
+2. Verify that they are uploaded properly:
+
+   ![files in project](/img/04.files.png)
+
+
+3. Navigate to the `App` section in the sidebar of the project and select a name for the Domino Cost App that you prefer and you can easily identify. The standar environment and the smaller harware tier will be enought to run it.
```
Comment (ddl-olsonJD) [inline README.md:37]: of the project and `add a title` for the Domino Cost App that
The `standard` environment

File: README.md [hunk 5]
```diff
@@ -1 +1,79 @@
...
+
+   ![project's dashboard](/img/05.publishApp.png)
+
+4. After publishing it, we'll be redirected to the `App Status` pages.
```
Comment (ddl-olsonJD) [inline README.md:41]: `you`'ll be redirected

File: README.md [hunk 6]
```diff
@@ -1 +1,79 @@
...
+
+   ![project's dashboard](/img/01.projectsDashboard.png)
+
+2. Select a name and visibility that you prefer for this project.
+
+   ![create new project](/img/02.createNewProject.png)
+
+---
+# Publish the App 
+Now that we have a dedicated project to host our app. In order to publish it follow this steps:
```
Comment (ddl-olsonJD) [inline README.md:26]: Once a dedicated project has been created or selected to host our app.

File: README.md [hunk 7]
```diff
@@ -1 +1,79 @@
...
+
+3. Navigate to the `App` section in the sidebar of the project and select a name for the Domino Cost App that you prefer and you can easily identify. The standar environment and the smaller harware tier will be enought to run it.
+
+   ![project's dashboard](/img/05.publishApp.png)
+
+4. After publishing it, we'll be redirected to the `App Status` pages. 
+
+   ![project's dashboard](/img/06.runApp.png)
+   
+   We'll have to wait until the status changes to `running`.
```
Comment (ddl-olsonJD) [inline README.md:45]: Wait until the status changes to `running`.

File: README.md [hunk 8]
```diff
@@ -1 +1,79 @@
...
+
+   ![project's dashboard](/img/07.appStatus.png)
+
+   In this moment all the requeriments for the Domino Cost App to run will start to be installed and it will take a moment until the app is actually running.
```
Comment (ddl-olsonJD) [inline README.md:49]: It will take a moment for the dependencies to install, and the Domino Cost App to start running.

File: README.md [hunk 9]
```diff
@@ -1 +1,79 @@
...
+
+## Checking running status (optional)
+  If we want to verify the proper set up of our up will be able to do that from the app's user output. To be able to access them, just follow the next steps:
```
Comment (ddl-olsonJD) [inline README.md:52]: To verify that the app is properly setup, you can check the app's user output. To access them, follow the next steps:

File: README.md [hunk 10]
```diff
@@ -1 +1,79 @@
...
+  1. Click on `App versions` tab.
+
+   ![appVersions](/img/08.appVersions.png)
+
+  2. Click on the las newest version in the table and a side bar will be display in the right of the screen.
```
Comment (ddl-olsonJD) [inline README.md:57]: Click on the newest version in the

File: README.md [hunk 11]
```diff
@@ -1 +1,79 @@
...
+  
+  3. Click on `User Output` and you'll see a log showing the setup of the environment. Once the legend `Solara server is starting at http://0.0.0.0:8888`, your app is ready.
+
+   ![logs](/img/10.serverRunning.png)
+
+---
+
+# Accessing Domino Cost App
+
+After a moment that the status app is status is set as `running` you'll be able to access the app by clicking in the button `View App` in the UI.
```
Comment (ddl-olsonJD) [inline README.md:69]: After a moment that the app's status will be set to `running` and you will be able to access the app by clicking on the button `View App` in the UI.

File: app/domino_cost.py [hunk 12]
```diff
@@ -0,0 +1,386 @@
...
+
+# For interacting with the different scopes
+breakdown_options = ["Execution Type", "Top Projects", "User", "Organization"]
+breakdown_to_param = {
+    # "Execution Type": "dominodatalab_com_workload_type",
+    "Top Projects": "dominodatalab_com_project_name",
+    "User": "dominodatalab_com_starting_user_username",
+    "Organization": "dominodatalab_com_organization_name",
+}
+#breakdown_choice = sl.reactive(breakdown_options[0])
```
Comment (ddl-olsonJD) [inline app/domino_cost.py:48]: remove comment if not needed.

File: app/domino_cost.py [hunk 13]
```diff
@@ -0,0 +1,386 @@
...
+window_choice = sl.reactive(window_options[0])
+
+# TODO: This should be replaced with real values
+EXECUTION_COST_MAX = os.getenv("DOMINO_EXECUTION_COST_MAX", None)
+PROJECT_MAX_SPEND = os.getenv("DOMINO_PROJECT_MAX_SPEND", 8)
+ORG_MAX_SPEND = os.getenv("DOMINO_ORG_MAX_SPEND", 500)
+
+BREAKDOWN_SPEND_MAP = {"Top Projects": PROJECT_MAX_SPEND, "Organization": ORG_MAX_SPEND}
+# If the user changes the global filter by clicking on a bar in the breakdown chart
+# (lefthand chart), we want to change the breakdown to something else
```
Comment (ddl-olsonJD) [inline app/domino_cost.py:68]: What does this mean?

**Review by ddl-olsonJD — CHANGES_REQUESTED** [review]

File: app/domino_cost.py (hunk 13 above)
Comment (ddl-juan-cistaro) [reply app/domino_cost.py:68]: That was an existing comment of the previous code. Sorry for that

**Review by ddl-olsonJD — APPROVED** [review]
Please see the change requests
not sure about the _func_names as these. functions will not be called outside of this file,

**Pull request merged**